src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1063,10 +1062,9 @@ public virtual Task<IdentityResult> RemoveClaimAsync(TUser user, Claim claim)
                 throw new ArgumentNullException("roles");
             }
 
-            var userRoles = await userRoleStore.GetRolesAsync(user, CancellationToken);
             foreach (var role in roles)
```

Comment:
A LINQ query always executes as a database query in EF and things in memory won't have any influence on what entities are included in the results. See https://github.com/aspnet/EntityFramework/issues/2982 in the backlog. I think we should take the chance to add the .Distinct() while we are changing this.